### PR TITLE
Add: Cloudsmith Icon

### PR DESCRIPTION
**Pull request by lskillen**
### Checklist
  - [x] I optimized the icon with SVGO or SVGOMG
  - [x] The SVG `viewbox` is `0 0 24 24`

### Description

Adds the icon for [Cloudsmith](https://cloudsmith.io), a managed hosted Package Management service.

Official Branding Page:
https://cloudsmith.io/branding/

Cloudsmith exists in Font Awesome 5 as a brand icon (so parity would be great!):
https://fontawesome.com/icons/cloudsmith?style=brands

Thank you! :-)

*Full disclosure:* I work at Cloudsmith (background included below).

### CODE REVIEW

**Comment by lskillen** [comment]
Fixing the scaling - Please hold off until we fix it. Thank you. :-)

**Comment by birjolaxew** [comment]
My concern with this icon is mostly the popularity of the brand. Although Cloudsmith isn't primarily a website, so the Alexa rank isn't a deal-breaker, [cloudsmith.io's rank](https://www.alexa.com/siteinfo/cloudsmith.io) is 6595514 - for the sake of not adding everyone's hobby projects (and to give us a scope) we generally limit ourselves to [brands with rank 500k+](https://github.com/simple-icons/simple-icons/blob/develop/CONTRIBUTING.md#requesting-an-icon).

For a fairer comparison, Cloudsmith has 238 Twitter followers, while other low-popularity brands like Highly, Micro.blog or IconJar have 615, 573 and 1419 respectively. This seems a lot closer than the Alexa ranks would indicate, although it could be skewed by how many people Cloudsmith follow (which tends to result in a follow-back)

I don't generally like playing doorman, and we aren't keen on harshly cutting off at a certain threshold, but it would seem that this is more of a guerilla marketing effort to get Cloudsmith's name out there - and although I can appreciate the effort, I don't know if that is something we want on Simple Icons. At the same time Cloudsmith seems to be a legit company that has been active since 2016, so the risk of bloat that is usually connected with adding "small" projects is less of a concern.

Thoughts @ericcornelissen and @davidklebanoff ?

<sup>(on a sidenote, it would've been nice if you had disclosed explicitly that you are affiliated with Cloudsmith)</sup>

**Comment by lskillen** [comment]
Hey @birjolaxew - Thank you for reviewing!

First of all, you're absolutely right, I should have disclosed that I'm affiliated with Cloudsmith. I stupidly assumed that performing the PR from @cloudsmith-io would have signalled that, but you're right to point out that it should be explicit. So apologies for that, I'll make sure to include that in all company-related PRs in the future, regardless of the source repository.

It's true that Cloudsmith is a small-ish startup, but it isn't a hobbyist project. The motivation for us to add an icon to simple-icons is that (1) we like the simple-icons set a lot and (2) we're also in the middle of integrating with [shields.io](https://shields.io) to offer badges/shields (as per requests from our users) for things like latest package versions, number of downloads for package, license used for packages. The Alexa ranking is very variable for us because most requests don't hit our site (we're distributing packages for downloads) - In saying that, our "best" is nowhere near 500k just yet, but working on it.

I also appreciate the angle about guerrilla marketing, but as per above ^ that isn't something that we're trying to achieve here. Although I would love that to be true, I really doubt that an icon link would generate the kind of traffic that would involve people looking for package management. It would however make our integration with shields much more pleasant. What we will do though is link back to simple-icons and state that we use the icon here for the shields badges, so the inverse might be true!

Happy to discuss any other part - Also happy to figure out a workaround and come back again at a later stage once we've achieved from traction. Thank you again, you're awesome. :-)

**Comment by ericcornelissen** [comment]
Even though there are some valid arguments in favor of adding the icon in the discussion so far, I think we should not (**yet**) add this icon. Both the Alexa ranking and Twitter followers are in my opinion just too far off.

> For a fairer comparison, Cloudsmith has 238 Twitter followers, while other low-popularity brands like Highly, Micro.blog or IconJar have 615, 573 and 1419 respectively. This seems a lot closer than the Alexa ranks would indicate, although it could be skewed by how many people Cloudsmith follow (which tends to result in a follow-back)

On that note, although I see why you would compare the Twitter followers I think we should not go that way. In contrast to the Alexa ranking, the Twitter followers (or Facebook likes or whatever...) requires people to use a service to show their appreciation. Taking myself as an example, I don't use Twitter and never like anything on Facebook, but my usage of certain brands is reflected in the Alexa ranking.

**Comment by birjolaxew** [comment]
> but my usage of certain brands is reflected in the Alexa ranking

@ericcornelissen Except in instances where using the product doesn't mean visiting the website (as is the case for e.g. IconJar). In those cases Alexa ranks can still be an indicator (especially if above 500k), but if we mostly rely on Alexa we'd put non-website based brands at a serious disadvantage to website based brands.

I think the important bit is considering *why* we have a popularity cutoff; one part is giving ourselves a scope to limit ourselves to, but something I find more important is avoiding "bloat" in the form of icons that we have to potentially update or remove in the future. This tends to happen a lot more for small companies or hobby projects, where the effort of maintenance is greater than the value it gives (since it gives little value).

That being said, I do agree that in this case the brand is just a bit too far from our normal threshold.

**Comment by lskillen** [comment]
Appreciate the input folks! We'll look at the workarounds and try again in the future. Keep up the good work. :)

**Comment by ericcornelissen** [comment]
Thank you for your understanding @lskillen 👌 Again, we'd be glad to add Cloudsmith at a later point in time

**Comment by lskillen** [comment]
@ericcornelissen @birjolaxew @PeterShaggyNoble @runxel @davidklebanoff Perhaps time to re-consider?

Here are the metrics from today:

- [Alexa Rank: 886,387](https://www.alexa.com/siteinfo/cloudsmith.io)
- [Twitter Followers: 394](https://twitter.com/cloudsmith)
- [Investment Raised: ~$3 Million](https://www.crunchbase.com/organization/cloudsmith-io)

We're not going anywhere. :-)

We're also proud to say that we power distribution of Font Awesome, for their "Pro" variant which is only accessible to customers with a license. Most of these users do not directly visit the cloudsmith.io website but interact via tools such as npm.

I'll leave it in your capable hands to decide, but happy to provided further information.

**Comment by ericcornelissen** [comment]
My apologies @lskillen I missed this comment... I closed your other issue to keep the conversation in one place 🙂 

Unfortunately the Alexa rank is still not sufficient and I don't think can consider that Twitter follower count enough (though we don't have an official number for that yet). I'm personally not in favour of using "investment raised" as a measure of popularity (for various reasons, including that those numbers don't mean anything to me).

The last point you bring up is more interesting to me. If you can provide "usage" numbers with some "big" examples (such as Font Awesome) then I think we can make a better assessment. I guess a service like Cloudsmith does not necessarily get a lot of visits on it's own website even when it is commonly used.

**Comment by lskillen** [comment]
Hi @ericcornelissen - Completely understood, I'll rationalise but otherwise defer to you. 😄 

I think, for us, the rationale for investment raised is that it's primarily a counter-argument to the "hobbyist project" defense. The amount isn't important, more so that there is any at all. It's extremely hard to actually raise money, because you've convinced hardened thinking individuals (e.g. VCs) that you can return a 20x, 50x or 100x multiple of the amount raised. That would be impossible for a hobby project unless it's seen explosive growth. :-)

It's a shame that the social media and Alexa metrics don't quite capture our situation, but we understand the need to have some sort of anchor/baseline. We typically don't try to drive mass amounts of unqualified traffic to the site (which would massively improve the Alexa rank, but is otherwise a vanity metric) or ask/beg people to follow us on twitter (so follows tend to come about much more naturally, via interactions). Basically we try to stay out of our users' way.

I can't give specific statistics because they almost entirely relate to customers (although I'm happy to answer general questions), and I don't have permission to share specifics yet; but I can say that we're in the 9 figures for the number of requests that we service per month, in total. This is (relatively) almost entirely machine-based requests, but it's global distribution all over the world, to most countries (see [here](https://blog.cloudsmith.io/2019/12/20/by-the-numbers-2019-edition/) for more figures). It's more appropriate to view Cloudsmith as a CDN-like website/service than a user-oriented one; it's a "developer tool" primarily.

My question: How can we help you set a standard for measuring services like this?

**Comment by ericcornelissen** [comment]
> I think, for us, the rationale for investment raised is that it's primarily a counter-argument to the "hobbyist project" defense. The amount isn't important, more so that there is any at all. It's extremely hard to actually raise money, because you've convinced hardened thinking individuals (e.g. VCs) that you can return a 20x, 50x or 100x multiple of the amount raised. That would be impossible for a hobby project unless it's seen explosive growth. :-)

It **is** a counter-argument to the "hobbyist project" defense, but I'm not entirely convinced it is an argument for longevity. I believe there are plenty of startups with reasonable investment that don't survive or a long time. Another not uncommon scenario, I think, is when a startup is bought by a bigger company and the separate brand is thrown away entirely.


> It's a shame that the social media and Alexa metrics don't quite capture our situation, but we understand the need to have some sort of anchor/baseline. We typically don't try to drive mass amounts of unqualified traffic to the site (which would massively improve the Alexa rank, but is otherwise a vanity metric) or ask/beg people to follow us on twitter (so follows tend to come about much more naturally, via interactions). Basically we try to stay out of our users' way. For what it's worth, we have 637 followers on LinkedIn, which might be relevant, but uncertain.
> 
> I can't give specific statistics because they almost entirely relate to customers (although I'm happy to answer general questions), and I don't have permission to share specifics yet; but I can say that we're in the 9 figures for the number of requests that we service per month, in total. This is (relatively) almost entirely machine-based requests, but it's global distribution all over the world, to most countries (see here for more figures). It's more appropriate to view Cloudsmith as a CDN-like website/service than a user-oriented one; it's a "developer tool" primarily.
> 
> My question: How can we help you set a standard for measuring services like this?

We understand the limitation of social media and Alexa rankings as well and I understand you cannot share too much. I'm still a bit caution about allowing this. Consider also that allowing this more or less requires us to allow a bunch of other stuff as well (if their icons gets requested).

One thing that would be interesting to know is if your users want the icon to be included 🤔

**Comment by lskillen** [comment]
> It **is** a counter-argument to the "hobbyist project" defense, but I'm not entirely convinced it is an argument for longevity. I believe there are plenty of startups with reasonable investment that don't survive or a long time. Another not uncommon scenario, I think, is when a startup is bought by a bigger company and the separate brand is thrown away entirely.

Gotcha. Hoping that isn't us. 😂 Well, there isn't too much else I can say here, but I completely get your point. I already stated previously that we've been around since 2016 "officially" (new company and pivot), and since 2012 "unofficially" (previous company that morphed into this). The investment is for sustainable growth, and helps to guarantee us for at least another 2 years. The intention is to become a "brand" name.

> We understand the limitation of social media and Alexa rankings as well and I understand you cannot share too much. I'm still a bit caution about allowing this. Consider also that allowing this more or less requires us to allow a bunch of other stuff as well (if their icons gets requested).
> 
> One thing that would be interesting to know is if your users want the icon to be included 🤔

We're happy to wait until the Alexa figure hits 500k, to be honest because we'll naturally hit it sometime in the next few months. Although I'd really like to see it in the set, as it means we no longer have to work around it in our integration with shields.io, which is a bit of pain.

Speaking of which, arguably the biggest users of a Cloudsmith icon via simple icons would be via the shields.io service since it is directly integrated. At the moment, the URI for our badges is *much* longer than it needs to because we have to embed the image in it each time, instead of being able to link against simple icons. In saying that, we do have companies that link to us from their (usually internal) documentation, and although I can't tell if they are using simple icons, they _might_ be.

Another question I had was: I stated that our number of requests fulfilled was 9 figures (i.e. hundreds of millions), across 195 countries. That's the "downloads" side. Are there other specific statistics that you'd be interested in knowing that would help you to fulfill "alternative" criteria? I get that this is hard to reason about. 😁

**Comment by ericcornelissen** [comment]
> I already stated previously that we've been around since 2016 "officially" (new company and pivot), and since 2012 "unofficially" (previous company that morphed into this). The investment is for sustainable growth, and helps to guarantee us for at least another 2 years. The intention is to become a "brand" name.

I didn't remember that and went by a different number I found somewhere online in my previous comment (I think it said 2018). The fact that you're around for a longer time works in your favour.


> Speaking of which, arguably the biggest users of a Cloudsmith icon via simple icons would be via the shields.io service since it is directly integrated. At the moment, the URI for our badges is much longer than it needs to because we have to embed the image in it each time, instead of being able to link against simple icons. In saying that, we do have companies that link to us from their (usually internal) documentation, and although I can't tell if they are using simple icons, they might be.

Can you point us to some of those shields in use 🙃 


> Another question I had was: I stated that our number of requests fulfilled was 9 figures (i.e. hundreds of millions), across 195 countries. That's the "downloads" side. Are there other specific statistics that you'd be interested in knowing that would help you to fulfill "alternative" criteria? I get that this is hard to reason about. 😁

Sorry I didn't get back to that 😞 As you said this is hard to reason about. The only comparison I can make, really, is to other content delivery sites we are already providing. For example, [jsDeliver](https://www.jsdelivr.com/) has 9 figures download numbers for all of their top-10 packages. But I don't know if that is a fair comparison...

**Comment by lskillen** [comment]
> Can you point us to some of those shields in use 🙃

You're giving me a run for my money today. 😂 👍 

Most of the badge uses are private (even to us, we just see the requests, but can't tell where it came). Where it isn't private, GitHub is usually utilised but also doesn't provide us the referrer when used, so it's hard to tell where our badges are actually embedded. However, I know of a few public ones off the top of my head from interactions (but could do further digging):

- https://honeydb.io/downloads/
- https://github.com/CyberReboot/poseidon/

Other uses are via the Cloudsmith site itself, e.g.:

- https://cloudsmith.io/~isc/repos/kea-1-7/packages/detail/rpm/isc-kea/1.7.4-isc0012520200128102739.fc30/a=x86_64;d=fedora%252F30;t=1/

> Sorry I didn't get back to that 😞 As you said this is hard to reason about. The only comparison I can make, really, is to other content delivery sites we are already providing. For example, [jsDeliver](https://www.jsdelivr.com/) has 9 figures download numbers for all of their top-10 packages. But I don't know if that is a fair comparison...

Cloudsmith isn't exactly a CDN in the same sense. It's just the closest equivalent. jsDeliver is an order of magnitude above us, but serves multiple individual files publicly, whereas we tend to serve aggregate packages mostly privately. The difference is, we might serve one large package, for a build pipeline, but jsDeliver is serving 100 files, embedded on a website. In other words, jsDeliver is hit each time the using website is hit, but our usage is directly related to builds/delivery. Not disputing that they're bigger than us though!

**Comment by ericcornelissen** [comment]
> You're giving me a run for my money today. :joy: :+1:

Sorry, I'm just trying to get to a point where we can add Cloudsmith


> Most of the badge uses are private (even to us, we just see the requests, but can't tell where it came). Where it isn't private, GitHub is usually utilised but also doesn't provide us the referrer when used, so it's hard to tell where our badges are actually embedded. However, I know of a few public ones off the top of my head from interactions (but could do further digging):
> 
> - https://honeydb.io/downloads/
> - https://github.com/CyberReboot/poseidon/
>
> Other uses are via the Cloudsmith site itself, e.g.:
>
> - https://cloudsmith.io/~isc/repos/kea-1-7/packages/detail/rpm/isc-kea/1.7.4-isc0012520200128102739.fc30/a=x86_64;d=fedora%252F30;t=1/

Thanks for the pointers...

> Cloudsmith isn't exactly a CDN in the same sense. It's just the closest equivalent. jsDeliver is an order of magnitude above us, but serves multiple individual files publicly, whereas we tend to serve aggregate packages mostly privately. The difference is, we might serve one large package, for a build pipeline, but jsDeliver is serving 1 to N (hundreds?) files, embedded on a website (typically). In other words, jsDeliver is hit each time the using website is hit, but our usage is directly related to builds/delivery. Not disputing that they're bigger than us though!

... And thanks for the clarification :slightly_smiling_face: 

---

Based on all this information I would be okay with including the Cloudsmith icon. However, since the brand doesn't strictly meet our popularity threshold I want to ask for some second opinions from other @simple-icons/maintainers

**Comment by runxel** [comment]
> Based on all this information I would be okay with including the Cloudsmith icon. However, since the brand doesn't strictly meet our popularity threshold I want to ask for some second opinions from other @simple-icons/maintainers

It seems Cloudsmith is here to stay for a while. 
And even tho they don't _seem_ to be popular it looks more like a mismatch on our site. So I would be okay to add this one in. Let's see if we can reach a monodic result! :)

**Comment by PeterShaggyNoble** [comment]
Yeah, I'd agree that we could add this. Reopening for review.

Thanks for your patience, @lskillen, and all the additional information you provided.

**Comment by lskillen** [comment]
And thank you @ericcornelissen, @runxel and @PeterShaggyNoble for your balanced analysis! This is/was hard to reason about, and I greatly appreciate your patience as well while determining the fit outcome. :raised_hands:

**Comment by PeterShaggyNoble** [comment]
Oop! Forgot to actually review this 🤦‍♂

The `path` doesn't quite fill the canvas, @lskillen - it's `23.897`. Once that's fixed, this should be ready to merge in.

Normally, I'd suggest picking the first colour (`#12577e`) from a branding page when there's no clear preference stated but, as you're a representative of Cloudsmith, I think we're OK to go with your choice! 😆

**Comment by lskillen** [comment]
> 
> 
> Oop! Forgot to actually review this 🤦‍♂
> 
> The `path` doesn't quite fill the canvas, @lskillen - it's `23.897`. Once that's fixed, this should be ready to merge in.
> 
> Normally, I'd suggest picking the first colour (`#12577e`) from a branding page when there's no clear preference stated but, as you're a representative of Cloudsmith, I think we're OK to go with your choice! 😆

Ha! Thanks @PeterShaggyNoble. #187EB6 is our primary colour. :-)

I'm not quite an SVG/frontend person. Any suggestions on how best to fix the path?

File: icons/cloudsmith.svg [hunk 1]
```diff
@@ -0,0 +1 @@
+<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" role="img"><title>Cloudsmith icon</title><path d="M19.939 19.818a4.133 4.133 0 1 1-4.13-4.135 4.133 4.133 0 0 1 4.13 4.135zm0 0M15.904 8.334h.519c.98-.069 1.922-.455 2.637-1.207a4.126 4.126 0 0 0 0-5.856 4.126 4.126 0 0 0-5.856 0c-.974.98-1.334 2.282-1.17 3.516-.456 2.536-2.833 2.896-3.87 2.928a4.062 4.062 0 0 0-3.22 1.202 4.116 4.116 0 0 0 0 5.855 4.12 4.12 0 0 0 5.85 0 4.088 4.088 0 0 0 1.208-2.668v-.228c.095-3.087 2.764-3.51 3.902-3.542zm0 0"/></svg>
```
Comment (phatbhoy67) [inline icons/cloudsmith.svg]:
```suggestion
<svg role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" ><title>Cloudsmith icon</title><path d="M16.15 0a4.146 4.146 0 0 0-2.94 1.225c-.981.98-1.34 2.288-1.177 3.53-.458 2.548-2.843 2.908-3.889 2.94-1.176-.098-2.352.327-3.235 1.21a4.142 4.142 0 0 0 0 5.88 4.142 4.142 0 0 0 5.882 0A4.136 4.136 0 0 0 12 12.108v-.23c.097-3.104 2.777-3.529 3.92-3.561h.523c.98-.066 1.928-.458 2.647-1.21a4.142 4.142 0 0 0 0-5.88A4.146 4.146 0 0 0 16.15 0zm-.327 15.7a4.15 4.15 0 0 0-4.15 4.15 4.15 4.15 0 0 0 4.15 4.15 4.15 4.15 0 0 0 4.15-4.15 4.15 4.15 0 0 0-4.15-4.15z"/></svg>
```

**Comment by lskillen** [comment]
@phatbhoy67 Looks fantastic, thank you. Your help is hugely appreciated!

**Pull request merged**